<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BinMainWindow</class>
 <widget class="QMainWindow" name="BinMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>971</width>
    <height>210</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Битовые операции</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLineEdit" name="binInput">
        <property name="font">
         <font>
          <pointsize>18</pointsize>
         </font>
        </property>
        <property name="inputMask">
         <string>BBBB.BBBB.BBBB.BBBB.BBBB.BBBB.BBBB.BBBB</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="tasksBox">
        <property name="minimumSize">
         <size>
          <width>600</width>
          <height>0</height>
         </size>
        </property>
        <item>
         <property name="text">
          <string>Задание 1</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Задание 2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Поменять байты</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Максимальная степень 2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Найти p: 2^p &lt;= x &lt;= 2^(p+1)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>xor</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Циклический сдвиг</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Перестановка бит</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="output">
        <property name="font">
         <font>
          <family>Courier New</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="plainText">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="maximumSize">
           <size>
            <width>10</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>=</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="input">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="inputMask">
           <string>0000000000</string>
          </property>
          <property name="maxLength">
           <number>10</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="updateButton">
        <property name="text">
         <string>Обновить число</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLineEdit" name="lineEdit">
          <property name="inputMask">
           <string/>
          </property>
          <property name="maxLength">
           <number>32</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Выполнить</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>971</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
